lib/mnesia/src/mnesia_subscr.erl

```diff
@@ -163,6 +163,15 @@ report_table_event({subscribers, S1, S2}, Tab, Tid, _Obj, clear_table, _Old) ->
     What4  = {write, schema,  {schema, Tab, TabDef}, [], Tid},
     deliver(S2, {mnesia_table_event, What4});
 
+report_table_event({subscribers, S1, _S2}, Tab, Tid, Obj, clear_table, _Old) ->
+    %% Obj is a match pattern here.
+    %% Sending delete_object event is compatible with `mnesia_loader`,
+    %% that uses `db_match_erase/2` which actually removes records by pattern.
+    %% Extended event is omitted: it's possible to match and get `OldRecords`,
+    %% but the list can be quite large.
+    Standard = {delete_object, patch_record(Tab, Obj), Tid},
+    deliver(S1, {mnesia_table_event, Standard});
```

Comment:
This is the relevant place, where standard event is used internally: https://github.com/emqx/otp/blob/f899f860e2f8ff93bc5f1af3e6802a1d58ad1d5c/lib/mnesia/src/mnesia_loader.erl#L644
Sending delete_object event when match_delete is called is thus backward-compatible..